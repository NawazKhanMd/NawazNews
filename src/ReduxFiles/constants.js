export const actions = {
    loading: 'Loading',
    getTopNews: "Get_Top_News",
    getTopNewsSuccess: "Get_News_Success",
    getTopNewsFailure: "Get_News_Failure"
} 