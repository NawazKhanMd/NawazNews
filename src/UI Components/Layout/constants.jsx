import React from 'react';
export const AvailableReadModes = ["normal", "dim", "BnW", "night"];
export const ResumeLink = "https://drive.google.com/file/d/1W2hwgGX8c_kW5njby1uy2jD-Df-Ys7J0/view?usp=sharing";

export const Inspiration = () => (
    <div>
        <h2>Hey Guys,</h2>
        <br />
        <h5>First of all, I would love to Thank you for showing interest in my profile. </h5>
        <h5>I have been Creating Web & Mobile applications since 2015.  </h5>
        <h5>I am an Electronics and Communication background guy. After finishing my Bachelor's Education I had realized that I have a passion for Mobile & Web applications. </h5>
        <h5>I first started with Angular JS, Ionic & Cordova then shifted to React-Native and React-JS.</h5>
        <br />
        <h3>Assignment</h3>
        <br />
        <h5>Initially, I thought of using a UI library example: Material UI or ReactStrap, I have been using these two extensively in my day-to-day projects. after a while, I decided not to use them and use plain SCSS for all the components. </h5>
        <br />
        <h5>This is my Miro Board which I frequently use, I have put your Requirements PDF as the base of my WebApp and Took out color inspirations and Placements of objects.</h5>
        <br />
        <div id="miroBoard"></div>
        <br />
        <div className="centerAlign">
            <h5>I took out some of the 3d Objects to use in the website, Logo of the website is also inspired by a Chinese Design company</h5>

            <div id="inspire2"></div>
        </div>
        <br />
        <h5>The background is inspired by the Interior design of your office, I love mimicking real-world solutions in Digital space</h5>
        <br />
        <div id="inspire3"></div>
        <br />
        <h5>Last but not least, I found this perfect color pallet for the website and declared my variable.scss</h5>
        <br />
        <div id="inspire4"></div>
        <br />
        <h4>I love creating Websites and Mobile applications which I recently found out, Earlier I thought I only liked it but now after spending 6 years in Web and Mobile App development, I am Sure!.</h4>
        <br />
        <br />
        <h2> Thank you for reviewing my Code + UI</h2>

    </div >
);
