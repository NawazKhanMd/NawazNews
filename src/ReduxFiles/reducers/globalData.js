import { actions } from "../constants";

export const initialState = {
    loading: false
};


export const GData = (state = initialState, action) => {
    switch (action.type) {
        case actions.loading:
            return { ...state, loading: action.payload }
        default:
            return state
    }
}
