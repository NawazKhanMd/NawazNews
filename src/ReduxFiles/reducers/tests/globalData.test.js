import { GData, initialState } from '../globalData';
import { actions } from '../../constants';

describe('Home Reducer', () => {
    it('Handles default case', () => {
        const state = GData(undefined, {});
        expect(state).toEqual(initialState);
    });
    it('Handles loading update ', () => {
        const data =  true;
        const action = { type: actions.loading, payload: data };
        const state = GData(undefined, action);
        expect(state).toMatchSnapshot();
    });
    it('Handles getAll News save ', () => {
        const action = { type: actions.getTopNews};
        const state = GData(undefined, action);
        expect(state).toMatchSnapshot();
    });
    it('Handles getA News save ', () => {
        const data =  "someCode";
        const action = { type: actions.getANews, payload: data };
        const state = GData(undefined, action);
        expect(state).toMatchSnapshot();
    });
});