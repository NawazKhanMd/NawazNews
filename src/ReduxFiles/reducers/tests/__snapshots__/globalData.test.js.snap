// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Reducer Handles getA News save  1`] = `
Object {
  "NewsData": Object {},
  "TopNewsCodes": Array [],
  "loading": false,
}
`;

exports[`Home Reducer Handles getAll News save  1`] = `
Object {
  "NewsData": Object {},
  "TopNewsCodes": Array [],
  "loading": false,
}
`;

exports[`Home Reducer Handles loading update  1`] = `
Object {
  "NewsData": Object {},
  "TopNewsCodes": Array [],
  "loading": true,
}
`;
